src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -540,7 +540,8 @@ private static TypeSymbol MakeMergedTupleType(ArrayBuilder<DeconstructionVariabl
             var inferredPositions = tupleNames.SelectAsArray(n => n != null);
 
             var type = TupleTypeSymbol.Create(syntax.Location,
-                typesBuilder.ToImmutableAndFree(), locationsBuilder.ToImmutableAndFree(),
+                typesBuilder.ToImmutableAndFree().SelectAsArray(TypeMap.AsTypeSymbolWithAnnotations),
+                locationsBuilder.ToImmutableAndFree(),
```

Comment:
Where are the annotations coming from? Should there be a PROTOTYPE for the fact that they must be computed?

Reply:
As we discussed, `BoundExpression.Type` could be changed to `TypeSymbolWithAnnotations` to cover these cases. Or the nullability could be obtained from `BoundExpressionExtensions.IsNullable()`. For now, I've added a `// PROTOTYPE` comment.

Reply:
Changed to use `BoundExpressionExtensions.GetTypeAndNullability()`.